src/main/java/org/zalando/nakadi/controller/CursorsController.java

```diff
@@ -82,6 +93,11 @@ public CursorsController(final CursorsService cursorsService,
         if (cursors.isEmpty()) {
             throw new CursorsAreEmptyException();
         }
+
+        if (blacklistService.isSubscriptionConsumptionBlocked(subscriptionId, client.getClientId())) {
```

Comment:
This will add a call to DB for quite a busy endpoint. Let's optimize this call.

Reply:
Thanks for pointing out. I just replaced the call to `SubscriptionDbRepository` by a call to `SubscriptionCache`. This should also improve the performance when opening a stream to a subscription.